testing/unittests/esdltests.cpp

```diff
@@ -1527,9 +1551,540 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
+      <IsSet>abc</IsSet>
+      <DidntFail>abc</DidntFail>
+    </FullName>
+    <Friends>Jane,John,Jaap,Jessica</Friends>
+    <How>
+      <Did>
+        <We>
+          <Get>
+            <Here whoknows="abc">
+              <IDontKnow>abc</IDontKnow>
+              <CopyTokens>
+                <token>aaa</token>
+                <token>bbb</token>
+                <token>ccc</token>
+                <token>yyy</token>
+              </CopyTokens>
+              <CopyFriends>
+                <token>Jane</token>
+                <token>John</token>
+                <token>Jaap</token>
+                <token>Jessica</token>
+              </CopyFriends>
+              <CopyFields>
+                <CopyField>
+                  <First>Joe</First>
+                </CopyField>
+                <CopyField>
+                  <ID>GI101</ID>
+                </CopyField>
+                <CopyField>
+                  <ID>GI102</ID>
+                </CopyField>
+              </CopyFields>
+              <CopyFullName>
+                <FullerName valid="true">
+                  <First>Joe</First>
+                  <ID>GI101</ID>
+                  <ID>GI102</ID>
+                </FullerName>
+              </CopyFullName>
+            </Here>
+          </Get>
+        </We>
+      </Did>
+    </How>
+  </Person>
+</root>)!!";
+
+
+        try {
+
+            Owned<IEspContext> ctx = createEspContext(nullptr);
+            Owned<IEsdlScriptContext> scriptContext = createTestScriptContext(ctx, input, config1);
+            runTransform(scriptContext, script, ESDLScriptCtxSection_ESDLRequest, "FirstPass", "target element 1", 0);
+
+            StringBuffer output;
+            scriptContext->toXML(output, "FirstPass");
+            if (result && !areEquivalentTestXMLStrings(result, output.str()))
+            {
+                fputs(output.str(), stdout);
+                fflush(stdout);
+                throw MakeStringException(100, "Test failed(%s)", "target element");
+            }
+        }
+        catch (IException *E)
+        {
+            StringBuffer m;
+            fprintf(stdout, "\nTest(%s) Exception %d - %s\n", "target element", E->errorCode(), E->errorMessage(m).str());
+            E->Release();
+            CPPUNIT_ASSERT(false);
         }
     }
+
+    void testHTTPPostXml()
+    {
+        static constexpr const char * input = R"!!(<?xml version="1.0" encoding="UTF-8"?>
+         <root>
+            <Person>
+               <FullName>
+                  <First>Joe</First>
+                  <ID>GI101</ID>
+                  <ID>GI102</ID>
+               </FullName>
+            </Person>
+          </root>
+        )!!";
+
+        static constexpr const char * script = R"!!(<es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Person">
+            <es:variable name='value' select="'abc'"/>
+            <es:http-post-xml url="'http://127.0.0.1:9876'" section="logging" name="roxiestuff">
```

Comment:
Is it valid to assume this service is published on the localhost every time the test is run?

Reply:
I don't currently include this test in the run.  It's there for Terrence and I, and to be run explicitly for test when set up correctly.  I'd like to get to the point where we have a way to run this type of test.